test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperCheckboxTest.cs

```diff
@@ -649,6 +649,54 @@ public void CheckBoxFor_WithComplexExpressions_DoesNotUseValuesFromViewDataDicti
             Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(html));
         }
 
+        [Fact]
+        public void CheckboxHelpers_UsesSpecifiedExpression()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1");
```

Comment:
There are existing checkbox tests that test all scenarios that you mentioned above (not just the `Assert.Throws` case). I'll add these cases for `RadioButton`.